### ISSUE TITLE
Find out why general purpose register can't hold large integer like 4567

### ISSUE THREAD

**Issue by justinhester** [justinhester/IntroCompOrg-RPi#12]
Listing 11.2.2 generates assembly code that only stores a word address to y = 4567, because it "doesn't fit" a ~2<sup>12</sup> integer in a CPU register. Yet, the subsequent assembly instruction

```
add r3, r4, r5
```

stores the result of 123 + 4567 in CPU register `r3`. Also, Chapter 8 section 2 discusses registers as 32-bit and/or 64-bit memory. So, which is it: 8-bit registers or 32-bit/64-bit registers?

**Comment by justinhester** [comment]
I opened up `assignment2.s` in gdb with the cli commands

```
$ as --gstabs -o assignment2.o assignment2.s
$ gcc -o assignment2 assignment2.o
$ gdb ./assignment2
```

I added a breakpoint at `ldr r4, yValue    @ y = 4567;` and the r4 register had an initial value of 0x104a8 (66728). I ran the gdb command `si` and the new value of r4 became 0x11d7 (4567). So, it seems the word at label .L3 didn't set r4 to a memory address, but rather the explicit value of y = 4567.
